### PR TITLE
FM-377 - Login logic for account details

### PR DESCRIPTION
**Pull request by ManFromMons**
Improved layout of the address section if there are postcode errors
Corrected the path to take the user to the correct edit screen

### CODE REVIEW

File: app/views/facilities_management/beta/buyer_details/edit.html.erb [hunk 1]
```diff
@@ -53,7 +53,7 @@
 
         <div class="govuk-form-group govuk-!-margin-top-4 <%= "govuk-form-group--error" if f.object.errors[:organisation_address_postcode].any? %>">
           <div class="">
-            <div id="fm-postcode-lookup-container" class="<%= ' govuk-visually-hidden' if f.object.organisation_address_line_1.present? %>">
+            <div id="fm-postcode-lookup-container" class="<%= ' govuk-visually-hidden' if f.object.organisation_address_line_1.present? && f.object.errors[:organisation_address_postcode].blank? %>">
```
Comment (iceraluk) [inline app/views/facilities_management/beta/buyer_details/edit.html.erb]: is this needed?

File: app/views/facilities_management/beta/buyer_details/edit.html.erb [hunk 2]
```diff
@@ -92,7 +92,7 @@
                 <a id="fm-building-not-found" href="new-building-address" class="govuk-link--no-visited-state" role="button"><%= t('.cant_find_addr_in_list') %></a>
               </div>
             </div>
-            <div class="govuk-!-margin-top-3<%= ' govuk-visually-hidden' unless f.object.organisation_address_line_1.present? %>" id="selected-address-container">
+            <div class="govuk-!-margin-top-3<%= ' govuk-visually-hidden' unless f.object.organisation_address_line_1.present? && f.object.errors[:organisation_address_postcode].blank? %>" id="selected-address-container">
```
Comment (iceraluk) [inline app/views/facilities_management/beta/buyer_details/edit.html.erb]: is this needed?

Comment (ManFromMons) [reply app/views/facilities_management/beta/buyer_details/edit.html.erb]: This one is arguably not needed....

Comment (ManFromMons) [reply app/views/facilities_management/beta/buyer_details/edit.html.erb]: No, in fact, they and another change are required....

File: app/views/facilities_management/beta/buyer_details/edit.html.erb [hunk 3]
```diff
@@ -92,14 +92,14 @@
...
             </div>
-            <div class="govuk-!-margin-top-3<%= ' govuk-visually-hidden' unless f.object.organisation_address_line_1.present? %>" id="selected-address-container">
+            <div class="govuk-!-margin-top-3<%= ' govuk-visually-hidden' unless f.object.organisation_address_line_1.present?%>" id="selected-address-container">
               <label class="govuk-heading-s govuk-!-margin-bottom-1"><%= t('.postcode') %></label>
               <div id="selected-address-postcode" class="govuk-!-margin-bottom-6">
                 <%= f.object.organisation_address_postcode %>
               </div>
               <label class="govuk-heading-s govuk-!-margin-bottom-1"><%= t('.selected_addr') %></label>
               <div id="selected-address-label" class="govuk-!-margin-bottom-6">
-                <%= f.object.organisation_address_line_1 + ', ' + f.object.organisation_address_line_2 + ', ' + f.object.organisation_address_town + ', ' + f.object.organisation_address_county + ', ' + f.object.organisation_address_postcode  if f.object.organisation_address_postcode.present?%>
```
Comment (iceraluk) [inline app/views/facilities_management/beta/buyer_details/edit.html.erb]: please leave this in

Comment (ManFromMons) [reply app/views/facilities_management/beta/buyer_details/edit.html.erb]: :-)

File: app/views/facilities_management/beta/buyer_details/edit.html.erb (hunk 1 above)
Comment (ManFromMons) [reply app/views/facilities_management/beta/buyer_details/edit.html.erb]: this is removed

**Pull request merged**